services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java

```
@@ -0,0 +1,138 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.event.service;
+
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchStatus;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SwitchService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SwitchService.class);
+
+    private TransactionManager transactionManager;
+    private SwitchRepository switchRepository;
+
+    public SwitchService(TransactionManager transactionManager, RepositoryFactory repositoryFactory) {
+        this.transactionManager = transactionManager;
+        this.switchRepository = repositoryFactory.createSwitchRepository();
+    }
+
+    /**
+     * Create or update switch in neo4j.
+     *
+     * @param sw switch.
+     * @return transaction was successful or not.
+     */
+    public boolean createOrUpdateSwitch(Switch sw) {
+        logger.info("Create or update switch {}", sw.getSwitchId());
+        transactionManager.begin();
+        try {
+            processCreateOrUpdateSwitch(sw);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
```

An exception must not be swallowed without a reason.

---

Fixed